pkg/package-server-manager/controller.go

```diff
@@ -98,22 +99,35 @@ func reconcileCSV(log logr.Logger, image string, csv *olmv1alpha1.ClusterService
 	if csv.ObjectMeta.CreationTimestamp.IsZero() {
 		log.Info("attempting to create the packageserver csv")
 	}
-	if !validateCSV(log, csv) {
-		log.Info("updating invalid csv to use the default configuration")
-		tmp, err := manifests.NewPackageServerCSV(
-			manifests.WithName(csv.Name),
-			manifests.WithNamespace(csv.Namespace),
-			manifests.WithImage(image),
-		)
-		if err != nil {
-			return err
-		}
-		csv.Spec = tmp.Spec
+
+	// The Package Server CSV is defined within this repo as a yaml manifest which
+	// acts as a "base" of the CSV we expect. We then apply the highAvailability
+	// configurations with the `ensureCSV` method to the expectedCSV object. The
+	// csv passed into this function then has the expected labels, annotations,
+	// and spec set.
+	expectedCSV, err := manifests.NewPackageServerCSV(
+		manifests.WithName(csv.Name),
+		manifests.WithNamespace(csv.Namespace),
+		manifests.WithImage(image),
+	)
+	if err != nil {
+		return err
 	}
-	if !ensureCSV(log, image, csv, highAvailabilityMode) {
+
+	if !ensureCSV(log, image, expectedCSV, highAvailabilityMode) {
 		log.V(3).Info("no further updates are necessary to the packageserver csv")
 	}
 
+	for k, v := range expectedCSV.GetLabels() {
+		csv.ObjectMeta.Labels[k] = v
+	}
+
+	for k, v := range expectedCSV.GetAnnotations() {
+		csv.ObjectMeta.Annotations[k] = v
+	}
+
+	csv.Spec = expectedCSV.Spec
```

Comment:
Grr.... I wanted to post my understanding as a suggestion, so you could easily tell me to punt if I'm off-base here.  
Where I'm not clear is if all the action is in the `ensureCSV` method, and action includes scope over all the accumulations we're making.  
It /seems/ so, in which case the last thing we want to do is invoke `ensureCSV`. 

Here's my best attempt at listing what I think should happen here:

```

	// The Package Server CSV is defined within this repo as a yaml manifest which
	// acts as a "base" of the CSV we expect. We then apply the highAvailability
	// configurations with the `ensureCSV` method to the expectedCSV object. The
	// csv passed into this function then has the expected labels, annotations,
	// and spec set.
	expectedCSV, err := manifests.NewPackageServerCSV(
		manifests.WithName(csv.Name),
		manifests.WithNamespace(csv.Namespace),
		manifests.WithImage(image),
	)
	if err != nil {
		return err
	}

	for k, v := range expectedCSV.GetLabels() {
		csv.ObjectMeta.Labels[k] = v
	}

	for k, v := range expectedCSV.GetAnnotations() {
		csv.ObjectMeta.Annotations[k] = v
	}

	csv.Spec = expectedCSV.Spec
	
	if !ensureCSV(log, image, csv, highAvailabilityMode) {
		log.V(3).Info("no further updates are necessary to the packageserver csv")
	}
	```

Reply:
Hey there @grokspawn, thanks for the review! I've moved all the logic within the ensure method so that it:
- Generates the csv from the manifest defined in code
- Applies HA configurations to the manifest
- Ensures that the proper annotations/labels exists
- That the actual spec matches the expected spec